src/util/StringUtils.h

```diff
@@ -207,6 +207,27 @@ inline size_t findLiteralEnd(const std::string_view input,
   return endPos;
 }
 
+// Implementation based on https://stackoverflow.com/a/25374036
+inline bool constantTimeEqualsImpl(std::basic_string_view<volatile char> str1,
+                            std::basic_string_view<volatile char> str2) {
+  if (str1.length() != str2.length()) {
+    return false;
+  }
+  volatile char c = 0;
+  for (size_t i = 0; i < str1.length(); ++i) {
+    c = c | (str1[i] ^ str2[i]);
```

Comment:
Interesting: Compound assignment for volatile variables is deprecated in C++20 and un-deprecated in C++23.
See https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1152r0.html#compound and https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2327r1.pdf for more information

Reply:
`mismatch = mismatch  || (str1[i] != str2[i])` does definitely the wrong thing because `||` is a short circuit operation, so if this would be used as soon as 2 chars mismatch the chars would no longer be read resulting in faster execution.
I'm not sure what the compiler does when swapping both arguments, so `mismatch` is evaluated last, but I fear there might be some processor optimization that makes branch prediction execute the loop faster even with this setup. So the key to success here is not to have any branches that depend on the input data (except for length). That's why I assume using bool might be problematic here.

You can have a look at a variant of the code here: https://godbolt.org/z/zP7a9sEhv
It uses `volatile std::byte` instead of `volatile char` at the cost of more explicit conversion. Regardless it does not fix the sonarcloud issue complaining about the general usage of volatile (which is legitimate in this niche use case, so it should likely get suppressed).